cubed/core/plan.py

```diff
@@ -195,6 +195,25 @@ def visualize(
             "fontsize": "10",
         }
         dag.graph["node"] = {"fontname": "helvetica", "shape": "box", "fontsize": "10"}
+
+        # do an initial pass to extract array variable names from stack summaries
+        array_display_names = {}
+        for (n, d) in dag.nodes(data=True):
+            if "stack_summaries" in d:
+                stack_summaries = d["stack_summaries"]
+                first_cubed_i = min(
+                    i for i, s in enumerate(stack_summaries) if s.is_cubed()
+                )
+                caller_summary = stack_summaries[first_cubed_i - 1]
+                array_display_names.update(caller_summary.array_names_to_variable_names)
+        # add current stack info
+        frame = inspect.currentframe().f_back  # go back one in the stack
+        stack_summaries = extract_stack_summaries(frame, limit=10)
+        first_cubed_i = min(i for i, s in enumerate(stack_summaries) if s.is_cubed())
+        caller_summary = stack_summaries[first_cubed_i - 1]
+        array_display_names.update(caller_summary.array_names_to_variable_names)
```

Comment:
Can this all be refactored out into a function?

Reply:
Yes, absolutely